apps/aecore/lib/aecore/txs/pool/worker.ex

```diff
@@ -78,4 +115,83 @@ defmodule Aecore.Txs.Pool.Worker do
     {:reply, tx_pool, %{}}
   end
 
+  @doc """
+  A function that adds a merkle proof for every single transaction
+  """
+
+  @spec add_proof_to_txs(list()) :: list()
+  def add_proof_to_txs(user_txs) do
+    for tx <- user_txs do
+      block = get_block_by_txs_hash(tx.txs_hash)
```

Comment:
I just noticed you are using `get_block_by_txs_hash(tx.txs_hash)` when you can use the much faster `Chain.get_block(tx.block_hash)`

Reply:
The Chain.get_block() function works only with the prev_hash, so it wouldn't work well in this case.